### PR TITLE
chore: set max SDK version to <3.0.0 and fix Dart 2 type error

### PR DESCRIPTION
**Pull request by chalin**
1. Adjusted pubspec constraints.
2. Made the equivalent of the following fixes to http_unmodifiable_map.dart: https://github.com/dart-lang/shelf/pull/101, https://github.com/dart-lang/shelf/pull/105. W/o these fixes the code doesn't compile, for details see https://github.com/dart-lang/sdk/issues/32542
3. Re-enabled VM tests.
4. Left browser tests disabled because of Browser tests result in IsolateSpawnException: Unable to spawn isolate #178.

cc @kwalrath @kevmoo

### CODE REVIEW

**Comment by zoechi** [comment]
Is Dart stable enough to use `<3.0.0`?
I thought this is supposed to be delayed until Dart stable is released.

**Comment by kwalrath** [comment]
@zoechi 2.0 stable won't allow packages without `<3.0.0`, so we need to update packages now, before the stable release. (Announcement will be coming soon.)

File: pubspec.yaml [hunk 1]
```diff
@@ -1,19 +1,24 @@
 name: http
-version: 0.12.0
-author: "Dart Team <misc@dartlang.org>"
-homepage: https://github.com/dart-lang/http
+version: 0.12.0+1
```
Comment (chalin) [inline pubspec.yaml]: @nex3 - shall I set the version to 1.0.0-dev?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,8 @@
+## 0.12.0+1
```
[the remark this answers is not in the document]
Comment (chalin) [reply CHANGELOG.md]: Ok, I'll remove this entry. Is there anything else to be done?

**Comment by sroddy** [comment]
Is there any hope to see this released as 1.0.0 any time soon?
I think that there are some issues with the master branch as most of the changes and features that were there some days ago (e.g. Middleware) do not exist anymore. It seems like it has been rewritten with the contents of the 0.11 branch

**Comment by kevmoo** [comment]
@sroddy – sadly, no. We need to dig in with this package and find a way to evolve the API more slowly.

We have a LOT of packages that depend on pkg:http – so we can't be huge breaking changes in a single release.

**Comment by sroddy** [comment]
@kevmoo, sorry to bother again, do you have at least an high-level idea about what are the next steps for the evolution of this plugin? We have an important amount of code in our apps that relies on `Middleware` and that feature is so important that it led us to manage our own forks of different codebases (including Flutter itself for some time) that are using this library in order to support the new version.
That code disappeared from one day to the other from the master branch.
What do you recommend to do? Write our own implementation of the Middleware and HandlerClient? Switch to the `experimental` branch? Fork this code?

Thanks for your time

**Comment by natebosch** [comment]
> What do you recommend to do? Write our own implementation of the Middleware and HandlerClient? Switch to the experimental branch? Fork this code?

If the code _using_ it is going to be published, probably fork and publish the package under a new name.
If it's not going to be published, and you were previously using a git dependency, switch the ref to the experimental branch is probably OK - but in general relying on a branch isn't a super safe thing to do. Better would be to add a SHA as the ref so you can be sure you're going to get the same version on each pub get.

**Comment by donny-dont** [comment]
@sroddy I forked the experimental branch to https://github.com/http-dart/http-next and will be publishing that in the near future. Just doing some cleanup around it.

**Pull request closed without merging**